ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/DefaultSynchronizer.java

```diff
@@ -114,6 +134,22 @@ public void start() {
     }
   }
 
+  private Path getStateQueueDirectory(final Path dataDirectory) {
+    Path queueDataDirectory = dataDirectory.resolve("fastsync/statequeue");
+    File queueDataFile = queueDataDirectory.toFile();
+    queueDataFile.mkdirs();
+    // Clean up this data for now (until fast sync resume functionality is in place)
+    queueDataFile.deleteOnExit();
```

Comment:
I don't think `deleteOnExit` will work for a directory that we'll wind up creating files inside.  The naming of `queueDataDirectory` and `queueDataFile` is a bit misleading here as well since they're both actually a directory just of type `Path` or `File`.  Not quite sure what the right naming is but I think I'd have `queueDataDirectory` and `queueDataDir` since they're the same thing in slightly different forms.

Reply:
Yeah, you're right - this won't clean up non-empty directories.  I'm still testing this locally, so I'll clean it up a bit and ping you when its ready to look at.

Reply:
Added a shutdown hook for the cleanup: https://github.com/PegaSysEng/pantheon/pull/746/commits/bf2306fd9e6b9ed95a60f7c332affdca3afe4676